service/history/workflow/task_refresher.go

```diff
@@ -157,6 +157,11 @@ func (r *TaskRefresherImpl) refreshTasksForWorkflowStart(
 	taskGenerator TaskGenerator,
 ) error {
 
+	executionState := mutableState.GetExecutionState()
+	if executionState.Status != enumspb.WORKFLOW_EXECUTION_STATUS_RUNNING {
```

Comment:
This assumes `GenerateWorkflowStartTasks` is no-op when workflow is not running, which may not be true in the future I think. Maybe add some comments for GenerateWorkflowStartTasks method, so that people can know if that logic is modified, they'll need to double check the logic here.

Reply:
i would agree with ^

Reply:
Synced offline. I am going to revert the change in refreshTasksForWorkflowStart